lib/action_controller/serialization.rb

```diff
@@ -23,7 +23,7 @@ def serialization_scope(scope)
     end
 
     def namespace_for_serializer
-      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
+      @namespace_for_serializer ||= self.class.module_parent unless self.class.module_parent == Object
```

Comment:
```suggestion
      @namespace_for_serializer ||= namespace_for_class(self.class) unless namespace_for_class(module_parent) == Object
    end
    
    if Module.method_defined?(:parent)
      def namespace_for_class(klass)
        klass.parent
      end
    elsif Module.method_defined?(:module_parent)
      def namespace_for_class(klass)
        klass.module_parent
      end
    else
      fail "Module must define either parent or module_parent"
    end
```

Reply:
I mean it could be a utility, but in any case, it's just ducky, and only runs once at runtime.  Other gems I've worked on like RSpec are full of this stuff

Reply:
I changed `namespace_for_class(module_parent)` to `namespace_for_class(self.class)`